Removing the X509Chain shim from System.Net.Security

CIPop opened the pull request:
Adding code proposed by @bartonjs and remving the X509CertificateExtensions shim.
Changing the test server and client certificates (pending https://github.com/dotnet/corefx-testdata/pull/3) and adding a few e2e tests.

Related: #3114.

CIPop commented:
___Important___: This is expected to fail tests until dotnet/corefx-testdata#3 is merged and built.

davidsh commented on src/System.Net.Security/tests/FunctionalTests/CertificateValidationClientServer.cs:
```diff
@@ -0,0 +1,120 @@
+using System.Net.Sockets;
+using System.Net.Test.Common;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
```
need blank line between top-level 'using' groups.

CIPop replied:
Done.

stephentoub commented:
@dotnet-bot test this please (GitHub timeout)

stephentoub commented:
@dotnet-bot test this please (GitHub timeouts)

CIPop commented:
@stephentoub as I mentioned this won't work until dotnet/corefx-testdata#3 is merged and built. There is no point in retrying the tests.

I'm now investigating on why the build system keeps timing out. Currently, I believe that the timeouts that we have in place for the failure case are too long and add-up to go over the allowed total execution time for the tests. 

I'm also investigating why corefx-test didn't build the package.

CIPop commented:
dotnet/corefx-testdata is not automatically uploading the packages. I have added the new System.Net.TestData package to MyGet and corrected the original sources within dotnet/corefx-testdata#4.

I've also fixed this PR by updating `project.json`.

CIPop commented:
All external dependencies have been correctly added. Tests have passed.

@davidsh @stephentoub PTAL

davidsh commented on src/Common/tests/System/Net/Capability.Security.cs:
```diff
@@ -0,0 +1,38 @@
+        private static Lazy<bool> trustedCertificateSupport = 
+            new Lazy<bool>(IsTrustedCertificateSupported, LazyThreadSafetyMode.ExecutionAndPublication);
+        
+        public static bool TrustedCertificate()
```
I'm not sure `TrustedCertificate` is the best word choice. First, it's really checking for a Trusted Root Certificate and not an end-entity server certificate that happens to be trusted. Second, I think this method need a verb.

Maybe `IsTrustedRootInstalled` would be a better name?

CIPop replied:
I agree. I've changed the name.

davidsh commented on src/System.Net.Security/tests/FunctionalTests/CertificateChainValidation.cs:
```diff
@@ -0,0 +1,39 @@
+    {
+        public static void Validate(X509Certificate2Collection expected, X509Chain actual)
+        {
+            Debug.WriteLine("Validating certificate chain.");
```
It probably doesn't matter as much since it's test code and not product code.  But is the `Debug.WriteLine` temporary? Or is there a plan to add some way to turn it off and/or switch to another kind of logging, i.e. System.Net test ETW logging (that I thought we had already).

CIPop replied:
Yes, this was temporary. Removed.

bartonjs commented on src/System.Net.Security/tests/FunctionalTests/CertificateValidationClientServer.cs:
```diff
@@ -0,0 +1,121 @@
+                        TestConfiguration.DefaultSslProtocols,
+                        false);
+
+                    Task.WaitAll(new Task[] { clientAuthentication, serverAuthentication }, TestConfiguration.TestTimeoutSeconds * 1000);
```
Don't you want to assert that this succeeded, vs timed out?

CIPop replied:
Good catch. These were last minute changes. Will assert.

bartonjs commented on src/System.Net.Security/tests/FunctionalTests/CertificateValidationClientServer.cs:
```diff
@@ -0,0 +1,121 @@
+                Task clientConnect = clientConnection.ConnectAsync(serverEndPoint.Address, serverEndPoint.Port);
+                Task<TcpClient> serverAccept = server.AcceptTcpClientAsync();
+
+                Task.WaitAll(new Task[] { clientConnect, serverAccept }, TestConfiguration.TestTimeoutSeconds * 1000);
```
Shouldn't you check that this succeeded?

CIPop replied:
Will assert.

bartonjs commented on src/Common/tests/System/Net/Capability.Security.cs:
```diff
@@ -0,0 +1,38 @@
+{
+    public static partial class Capability
+    {
+        private const string CARootThumbprint = "3B279AD43D6DD459268D3F3A3D72DAAD4BF4D9C6";
```
Perhaps a comment suggesting what cert this matches?

CIPop replied:
Added a comment containing the Subject and the location of the p7b file containing the file.

bartonjs commented:
A few nitpicks in the tests, otherwise LGTM.

davidsh commented:
LGTM
